source/slang/check.cpp

```diff
@@ -1147,7 +1154,7 @@ namespace Slang
         {
             // TODO: we may want other cases here...
 
-            if (auto errorType = expr->type->As<ErrorType>())
+            if (auto errorType = expr->type.As<ErrorType>())
```

Comment:
These changes are worrying. I understand that these are around the null-`this` issues, but they crash head first into the fact that `Type::As` does *not* just do a `dynamic_cast`, and so it isn't equivalent to `RefPtr<Type>::As` unless you managed to customize that somehow.

Any place where we change a `Type::As` call to a `RefPtr::As` call is a potential bug, because it won't work correctly when the type just happens to be a `typedef`. I recently fixed a bug along these lines in the reflection code where we were using ordinary dynamic casts instead of the `Type::As` machinery, which ended up breaking `RWStructuredBuffer<float2>` because `float2` is a `NamedExpressionType` and not a `VectorExpressionType`.